db/remote/transaction.go

```diff
@@ -0,0 +1,76 @@
+package remote
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/grpc/gen"
+	"github.com/NethermindEth/juno/utils"
+)
+
+var _ db.Transaction = (*transaction)(nil)
+
+type transaction struct {
+	client gen.KV_TxClient
+	log    utils.SimpleLogger
+}
+
+func (t *transaction) NewIterator() (db.Iterator, error) {
+	err := t.client.Send(&gen.Cursor{
+		Op: gen.Op_OPEN,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	pair, err := t.client.Recv()
+	if err != nil {
+		return nil, err
+	}
+
+	return &iterator{
+		client:   t.client,
+		cursorID: pair.CursorId,
+		log:      t.log,
+	}, nil
+}
+
+func (t *transaction) Discard() error {
+	return t.client.CloseSend()
+}
+
+func (t *transaction) Commit() error {
```

Comment:
Can we have a separate `ReadOnlyDB` interface, and then embed it in the `DB` interface? It should remove the need for these superfluous functions and make the db argument to `rpc.New` more precise.

Reply:
Looking at the subsequent commits, it looks like this wouldn't work well since the db is also passed to the synchronizer.